### PR TITLE
Switch adist to stringdist

### PR DESCRIPTION
**Pull request by ehwenk**
Switching from `util:adist` to `stringdist:stringdist` for matching. This is both much faster and allows us to use a more nuanced matching algorithm by implementing the Damerau–Levenshtein distance method, and prioritising types of string changes (based on their algorithm).

### CODE REVIEW

**Comment by ehwenk** [comment]
@dfalster I've run all 47,000 AusTraits names through this and there were 33 that were different - it seems they are all instances of names that were passed over during fuzzy matching (match 5's) previously and now are being caught. So some additional matching power, but nothing being misaligned.

File: R/align_taxa.R [hunk 1]
```diff
@@ -201,7 +201,7 @@ align_taxa <- function(original_name,
 
   perfect_matches <- taxa$tocheck %>%
     filter(original_name %in% resources$`APC list (accepted)`$canonical_name) %>%
-    distinct() %>%
+    distinct(original_name) %>%
```
Comment (dfalster) [inline R/align_taxa.R:204]: This was intended?

**Review by dfalster — APPROVED** [review]

File: R/fuzzy_match.R [hunk 2]
```diff
@@ -9,13 +9,16 @@
 #' @param max_distance_rel The maximum proportional difference between the input string and the match
 #' @param n_allowed The number of allowable matches returned. Defaults to 1
 #' @param epithet_letters A string specifying if 1 or 2 letters remain fixed at the start of the species epithet.
+#' @param select_first_match If there are multiple equidistant matches, a variable indicating the first should be used, to avoid
```
Comment (dfalster) [inline R/fuzzy_match.R:12]: Was this intended? I can't see it being used below

**Review by dfalster — APPROVED** [review]

File: R/fuzzy_match.R (hunk 2 above)
Comment (ehwenk) [reply R/fuzzy_match.R]: no, removed now - part of the scheme I was working on before Will alerted me to stringdist

File: R/align_taxa.R (hunk 1 above)
Comment (ehwenk) [reply R/align_taxa.R:204]: Yes - the first "number of taxa" to match is unique names and this wasn't, which didn't make sense. For instance, "You've submitted 15 names. Of these 60 are perfect matches."

**Pull request merged**